envs/quadrotors/quad_utils.py

```diff
@@ -48,6 +49,13 @@ def make_quadrotor_env_single(cfg, **kwargs):
 
 def make_quadrotor_env_multi(cfg, **kwargs):
     from gym_art.quadrotor_multi.quadrotor_multi import QuadrotorEnvMulti
+    global CUSTOM_ENCODER_INITIALIZED
+
+    # TODO: Make this less hacky
```

Comment:
Yeah, not a fan of this stuff myself. If you have a better idea, let me know :D
I think CUSTOM_ENCODER_INITIALIZED is not even required here, you can just check in register_models if the encoder is already registered, and skip if it is.

Reply:
What was the reasoning behind registering models and having an ENCODER_REGISTRY?

Reply:
This allows SampleFactory to be used as an external dependency.
So one can just install it as a pip package instead of hacking directly in the repo. And define their custom architectures to be used by SampleFactory.
See examples/train_custom_env_custom_model.py for details.

I think encoder registry makes sense (just a set of all supported encoders), just like environment registry.
What's redundant here is the CUSTOM_ENCODER_INITIALIZED variable.

Reply:
I don't know how to get rid of this variable since without it, I get assertion errors saying that the encoder is already in the registry. I refactored the code to make it the same as how the encoder is added in doom_utils.py. Do you have any suggestions for other ways of doing this?